store/iavl/store.go

```diff
@@ -275,31 +277,42 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
-		if req.Prove {
-			value, proof, err := tree.GetVersionedWithProof(key, res.Height)
-			if err != nil {
-				res.Log = err.Error()
-				break
-			}
-			if proof == nil {
-				// Proof == nil implies that the store is empty.
-				if value != nil {
-					panic("unexpected value for an empty proof")
-				}
+		_, res.Value = tree.GetVersioned(key, res.Height)
+		if !req.Prove {
+			break
+		}
+		// Continue to prove existence/absence of value
+		var commitmentProof *ics23.CommitmentProof
+		var err error
+
+		// Must convert store.Tree to iavl.MutableTree to use in CreateProof
```

Comment:
If we can use another type in ics23iavl, please let me know and make a pr to simplify this logic

Thank you for handling this case, but it really should go in that library (once you get it working here, let's pull it over there)